source/common/grpc/codec.cc

```diff
@@ -31,20 +31,24 @@ void Encoder::prependFrameHeader(uint8_t flags, Buffer::Instance& buffer, uint32
   buffer.prepend(frame_buffer);
 }
 
-bool Decoder::decode(Buffer::Instance& input, std::vector<Frame>& output) {
+absl::Status Decoder::decode(Buffer::Instance& input, std::vector<Frame>& output) {
   // Make sure those flags are set to initial state.
   decoding_error_ = false;
   is_frame_oversized_ = false;
   output_ = &output;
   inspect(input);
   output_ = nullptr;
 
-  if (decoding_error_ || is_frame_oversized_) {
-    return false;
+  if (decoding_error_) {
+    return absl::InternalError("Grpc decoding error");
```

Comment:
I don't see any explicit test for this behavior that validates the error being returned (internal error).

Reply:
I was using `EXPECT_FALSE(decoder.decode(buffer, frames).ok());` for convenience :).

Updated.